### PR TITLE
fix(pill-selector): expose onInput to trap input

### PR DESCRIPTION
**Pull request by priyajeet**
Also fixes a few flow types and defaulting

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @priyajeet has signed the CLA. Thanks for the pull request!

File: src/components/pill-selector-dropdown/PillSelectorDropdownField.js [hunk 1]
```diff
@@ -14,19 +14,21 @@ type Props = FieldProps & {
     /** CSS class for the component. */
     className?: string,
     /** Given selected values and input text, returns a list of filtered options. Defaults to defaultDropdownFilter. */
-    dropdownFilter?: (options: Array<Option>, selectedValues: Array<Option>, inputText: string) => Array<Option>,
+    dropdownFilter: (options: Array<Option>, selectedValues: Array<Option>, inputText: string) => Array<Option>,
```
Comment (jpfranco) [inline src/components/pill-selector-dropdown/PillSelectorDropdownField.js]: I think removing the `?` might make flow complain about a missing prop if the consumer doesn't provide a filter.

File: src/components/pill-selector-dropdown/PillSelectorDropdownField.js [hunk 2]
```diff
@@ -43,10 +45,12 @@ type State = {
 
 class PillSelectorDropdownField extends React.PureComponent<Props, State> {
     static defaultProps = {
+        dropdownFilter: defaultDropdownFilter,
```
Comment (jpfranco) [inline src/components/pill-selector-dropdown/PillSelectorDropdownField.js]: Ah yes, I thought I had updated this. Probably just set `defaultDropdownRenderer` and forgot about this one.

File: src/components/pill-selector-dropdown/PillSelectorDropdownField.js (hunk 1 above)
Comment (priyajeet) [reply src/components/pill-selector-dropdown/PillSelectorDropdownField.js]: Its part of defaultProps, so flow should be happy

Comment (jpfranco) [reply src/components/pill-selector-dropdown/PillSelectorDropdownField.js]: Oh ok, cool. Didn't know that flow used the `defaultProps`

**Pull request merged**